core/go/internal/groupmgr/manager.go

```diff
@@ -0,0 +1,646 @@
+	}
+
+	// Insert the transaction
+	txIDs, err := gm.txManager.SendTransactions(ctx, dbTX, tx.TX)
+	if err != nil {
+		return nil, err
+	}
+
+	// We have the privacy group, and the state, so we can store all of these in the DB transaction - along with a reliable
+	// message transfer to all the parties in the group so they get notification it's there.
+	dbPG := &persistedGroup{
+		ID:              id,
+		Created:         tktypes.TimestampNow(),
+		Domain:          spec.Domain,
+		SchemaID:        genesisSchemaID,
+		SchemaSignature: stateABIs[0].Signature(),
+		GenesisTX:       txIDs[0],
+	}
+	if err = gm.insertGroup(ctx, dbTX, dbPG, spec.Members); err != nil {
+		return nil, err
+	}
+	group = dbPG.mapToAPI()
+	group.Genesis = states[0].Data
+	group.Members = spec.Members
+
+	// We also need to create a reliable send the state to all the remote members
+	msgs := make([]*pldapi.ReliableMessage, 0, len(remoteMembers))
+	for node, members := range remoteMembers {
+		for _, identity := range members {
+			msgs = append(msgs, &pldapi.ReliableMessage{
+				Node:        node,
+				MessageType: pldapi.RMTPrivacyGroup.Enum(),
+				Metadata: tktypes.JSONString(&components.PrivacyGroupDistribution{
+					GenesisTransaction: txIDs[0],
+					GenesisState: components.StateDistributionWithData{
+						StateDistribution: components.StateDistribution{
+							IdentityLocator: identity,
+							Domain:          spec.Domain,
+							StateID:         id.String(),
+							SchemaID:        genesisSchemaID.String(),
+						},
+					},
+				}),
+			})
+		}
+	}
+	if len(msgs) > 0 {
+		if err := gm.transportManager.SendReliable(ctx, dbTX, msgs...); err != nil {
+			return nil, err
+		}
+	}
+
+	return group, nil
+}
+
+func (gm *groupManager) StoreReceivedGroup(ctx context.Context, dbTX persistence.DBTX, domainName string, tx uuid.UUID, schema *pldapi.Schema, state *pldapi.State) (rejectionErr, err error) {
+
+	// We need to call the domain to validate the state schema before we can insert it
+	domain, rejectionErr := gm.domainManager.GetDomainByName(ctx, domainName)
+	if rejectionErr != nil {
+		return rejectionErr, nil
+	}
+	members, rejectionErr := domain.ValidatePrivacyGroup(ctx, schema, state)
+	if rejectionErr != nil {
+		return rejectionErr, nil
+	}
+
+	// Now do the insert
+	dbPG := &persistedGroup{
+		ID:              state.ID,
+		Created:         state.Created,
+		Domain:          domainName,
+		SchemaID:        state.Schema,
+		SchemaSignature: schema.Signature,
+		GenesisTX:       tx,
+	}
+	return nil, gm.insertGroup(ctx, dbTX, dbPG, members)
+
+}
+
+func (gm *groupManager) enrichMembers(ctx context.Context, dbTX persistence.DBTX, pgs []*pldapi.PrivacyGroup) error {
+	if len(pgs) == 0 {
+		return nil
+	}
+	groupIDs := make([]tktypes.HexBytes, len(pgs))
+	for i, pg := range pgs {
+		groupIDs[i] = pg.ID
+	}
+	var dbMembers []*persistedGroupMember
+	err := dbTX.DB().WithContext(ctx).
+		Where(`"group" IN ( ? )`, groupIDs).
+		Order("domain").
+		Order(`"group"`).
```

Comment:
Why is this quoted like it is?

Reply:
Because `group` is a special SQL word that has to be escaped